gslib/commands/signurl.py

```diff
@@ -313,18 +323,29 @@ def _GenSignedUrl(key,
         signed_headers=signed_headers,
         billing_project=billing_project,
         string_to_sign_debug=string_to_sign_debug)
-    raw_signature = sign(key, string_to_sign, digest)
+    if isinstance(key, PKey):
+      raw_signature = sign(key, string_to_sign, digest)
+    else: 
+      if not HAVE_CRYPTO:
+        raise CommandException(_CRYPTO_IMPORT_ERROR)
+      raw_signature = key.sign(to_bytes(string_to_sign), padding.PKCS1v15(), hashes.SHA256())
```

Comment:
Could you tell me a bit more about why this `to_bytes` function is necessary? Is it possible for CreatePayload to return either a string or bytes?